addons/payment_nuvei/tests/test_payment_provider.py

```diff
@@ -0,0 +1,47 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import tagged
+
+from odoo.addons.payment_nuvei.tests.common import NuveiCommon
+
+
+@tagged('post_install', '-at_install')
+class TestPaymentProvider(NuveiCommon):
+
+    def test_incompatible_with_unsupported_currencies(self):
+        """ Test that Nuvei providers are filtered out from compatible providers when the currency
+        is not supported. """
+        compatible_providers = self.env['payment.provider']._get_compatible_providers(
+            self.env.company.id, self.partner.id, self.amount, currency_id=self.env.ref('base.AFN').id
+        )
```

Comment:
```suggestion
        currency_id = self.env.ref('base.AFN').id
        compatible_providers = self.env['payment.provider']._get_compatible_providers(
            self.env.company.id, self.partner.id, self.amount, currency_id=currency_id
        )
```
Nit to stay < 100 chars